components/AuctionHeader.tsx

```diff
@@ -21,10 +22,49 @@ type Props = {
   noun: Noun | null;
 };
 
+const abi = [
+  {
+    inputs: [
+      {
+        internalType: 'uint256',
+        name: 'nounId',
+        type: 'uint256',
+      },
+      {
+        internalType: 'uint32',
+        name: 'clientId',
+        type: 'uint32',
+      },
+    ],
+    name: 'createBid',
+    outputs: [],
+    stateMutability: 'payable',
+    type: 'function',
+  },
+];
+
 export default function AuctionHeader(props: Props) {
-  const { isConnected } = useAccount();
+  const { isConnected, chainId } = useAccount();
+  const { switchChainAsync } = useSwitchChain();
   const write = useWriteContract({});
 
+  const bidMutation = useMutation({
+    mutationFn: async (bid: bigint) => {
+      if (chainId !== 1) {
+        await switchChainAsync({ chainId: 1 });
+      }
+      await write.writeContractAsync({
+        __mode: 'prepared',
```

Comment:
Add a comment: we are using this to allow the user's wallet to simulate the transaction and show if they don't have enough money